### PR TITLE
dedupe names in test also ensure for live we get new names

### LINKED ISSUE #3012: Investigate test failures

### PR DESCRIPTION
**Pull request by gearama**
fixes #3012 
# Pull Request Checklist

Please leverage this checklist as a reminder to address commonly occurring feedback when submitting a pull request to make sure your PR can be reviewed quickly:

See the detailed list in the [contributing guide](https://github.com/Azure/azure-sdk-for-cpp/blob/main/CONTRIBUTING.md#pull-requests).

- [x] [C++ Guidelines](https://azure.github.io/azure-sdk/cpp_introduction.html)
- [x] Doxygen docs
- [x] Unit tests
- [x] No unwanted commits/changes
- [x] Descriptive title/description
  - [x] PR is single purpose
  - [x] Related issue listed
- [x] Comments in source
- [x] No typos
- [x] Update changelog
- [x] Not work-in-progress
- [x] External references or docs updated
- [x] Self review of PR done
- [x] Any breaking changes?

### CODE REVIEW

File: sdk/keyvault/azure-security-keyvault-certificates/test/ut/certificate_client_base_test.hpp [hunk 1]
```diff
@@ -302,5 +303,18 @@ namespace Azure {
...
       }
     }
+
+    std::string GetCertificateName(std::string const& name)
+    {
+      std::string returnVal(name);
+      // due to delays in delete and purge we need to have different names for certificates as it
+      // will cause conflicts when trying to create a certificate in the process of
+      // purge/delete
+      if (GetEnv("AZURE_TEST_MODE") == "LIVE")
```
Comment (vhvb1989) [inline sdk/keyvault/azure-security-keyvault-certificates/test/ut/certificate_client_base_test.hpp:313]: this is ugly. No test should ever check or know about LIVE or anything. 

Can we just give unique-name for each test-case?   You can use the name of the test for creating the resource, for instance.

We can remove the delete/purge for all tests where we are not testing that functionality.  The entire Key Vault will be removed as soon as the tests finishes, so we don't need to do create+test+delete+purge for every test case. We just need create+test

File: sdk/keyvault/azure-security-keyvault-certificates/test/ut/certificate_client_test.cpp [hunk 2]
```diff
@@ -19,8 +19,8 @@ using namespace std::chrono_literals;
 TEST_F(KeyVaultCertificateClientTest, CreateCertificate)
 {
   // cspell: disable-next-line
-  std::string const certificateName("magiqStuff");
-  m_defaultWait = 5s;
+  std::string const certificateName(GetCertificateName("magiqStuff"));
```
Comment (vhvb1989) [inline sdk/keyvault/azure-security-keyvault-certificates/test/ut/certificate_client_test.cpp:22]: use:
```cpp
std::string testName = ::testing::UnitTest::GetInstance()->current_test_info()->name();
auto const& client
      = GetClientForTest(testName);
```

And use the testName as the name for whatever resource you create.

**Review by vhvb1989 — COMMENTED** [review]
Suggesting another strategy.  A test should never check how it is running to take decisions.

File: sdk/keyvault/azure-security-keyvault-certificates/test/ut/certificate_client_base_test.hpp (hunk 1 above)
Comment (gearama) [reply sdk/keyvault/azure-security-keyvault-certificates/test/ut/certificate_client_base_test.hpp:313]: no, that will ef up the record/playback. if i give random names each time i cannot guarantee the same name at record

Comment (vhvb1989) [reply sdk/keyvault/azure-security-keyvault-certificates/test/ut/certificate_client_base_test.hpp:313]: > if i give random names each time

Don't.  Use the `testName` ( always the same name )

**Comment by gearama** [comment]
Live tests passed
image

**Review by vhvb1989 — APPROVED** [review]

**Comment by ahsonkhan** [comment]
Does this fix https://github.com/Azure/azure-sdk-for-cpp/issues/3009 too? Might be worth verifying with a couple of live test runs and closing :)

**Pull request merged**